src/main/java/io/github/bobocodebreskul/context/support/ReflectionUtils.java

```diff
@@ -1,16 +1,150 @@
+    Objects.requireNonNull(clazz, "The class parameter cannot be null!");
+    log.trace("Scanning class {} for@ @{} existence", clazz.getName(), annotation.getSimpleName());
     return clazz.isAnnotationPresent(annotation);
   }
 
+
+  /**
+   * Checks if a specified annotation is present on the given constructor.
+   *
+   * @param annotation  The annotation type to check for. Must not be null.
+   * @param constructor The constructor on which to check for the presence of the annotation. Must
+   *                    not be null.
+   * @return {@code true} if the specified annotation is present on the given constructor,
+   * {@code false} otherwise.
+   * @throws NullPointerException If either the {@code annotation} or {@code constructor} parameter
+   *                              is {@code null}.
+   */
+  public static boolean isAnnotationPresentForConstructor(Class<? extends Annotation> annotation,
+      Constructor<?> constructor) {
+    Objects.requireNonNull(annotation, "The annotation parameter cannot be null!");
+    Objects.requireNonNull(constructor, "The constructor parameter cannot be null!");
+    return constructor.isAnnotationPresent(annotation);
+  }
+
+
+  /**
+   * Checks if a specified annotation is present on any of the given constructors.
+   *
+   * @param annotation   The annotation type to check for. Must not be null.
+   * @param constructors The constructors on which to check for the presence of the annotation. Must
+   *                     not be null.
+   * @return {@code true} if the specified annotation is present on any of the given constructors,
+   * {@code false} otherwise.
+   * @throws NullPointerException If either the {@code annotation} or {@code constructors} parameter
+   *                              is {@code null}.
+   * @see #isAnnotationPresentForConstructor
+   */
+  public static boolean isAnnotationPresentForAnyConstructor(Class<? extends Annotation> annotation,
+      Constructor<?>... constructors) {
+    return Arrays.stream(constructors)
+        .anyMatch(constructor -> isAnnotationPresentForConstructor(annotation, constructor));
+  }
+
+  /**
+   * Checks if a specified annotation is present on only one of the given constructors.
+   *
+   * @param annotation   The annotation type to check for. Must not be null.
+   * @param constructors The constructors on which to check for the presence of the annotation. Must
+   *                     not be null.
+   * @return {@code true} if the specified annotation is present on only one of the given
+   * constructors, {@code false} otherwise.
+   * @throws NullPointerException If either the {@code annotation} or {@code constructors} parameter
+   *                              is {@code null}.
+   * @see #isAnnotationPresentForConstructor
+   */
+  public static boolean isAnnotationPresentForSingleConstructorOnly(
+      Class<? extends Annotation> annotation,
+      Constructor<?>... constructors) {
+    return Arrays.stream(constructors)
```

Comment:
why not reuse getConstructorsAnnotatedWith method?

Reply:
Could you please explain in more details how do you see it?